android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -1143,4 +1144,123 @@ class QuestionnaireViewModelTest : RobolectricTest() {
       bundle.entry.any { it.resource is Basic && it.resource.id == "basic-resource-id" },
     )
   }
+
+  @Test
+  fun testApplyRelatedEntityLocationMetaTagWithResourceIdentifier() = runBlocking {
+    val questionnaireConfig =
+      questionnaireConfig.copy(
+        resourceIdentifier = "resourceId",
+        resourceType = ResourceType.Observation,
+        saveQuestionnaireResponse = false,
+        type = "EDIT",
+        extractedResourceUniquePropertyExpressions =
+          listOf(
+            ExtractedResourceUniquePropertyExpression(
+              ResourceType.Observation,
+              "Observation.code.where(coding.code='obs1').coding.code",
+            ),
+          ),
+      )
+    val questionnaireResponse =
+      QuestionnaireResponse().apply {
+        id = "resourceId"
+        meta.lastUpdated = Date()
+        subject = patient.asReference()
+        questionnaire = samplePatientRegisterQuestionnaire.asReference().reference
+      }
+    val subjectType = ResourceType.Observation
+
+    assertEquals("resourceId", questionnaireResponse.id)
+    assertEquals(subjectType, questionnaireConfig.resourceType)
```

Comment:
These assertions will always be evaluated to be true. The assertions are not done against the meta tags as the name of the test suggests.